packages/cardano-graphql-services/src/DataProjection/blockHandlers/helpers.ts

```diff
@@ -0,0 +1,79 @@
+import { Asset, BigIntMath } from '@cardano-sdk/core';
+import { AssetMetadata } from '../../MetadataClient/types';
+import { Schema, isAlonzoBlock, isMaryBlock } from '@cardano-ogmios/client';
+
+export const getBlockType = (block: Schema.Block) => {
+  let b: Schema.BlockMary | Schema.BlockAlonzo | undefined;
+  if (isAlonzoBlock(block)) {
+    b = block.alonzo as Schema.BlockAlonzo;
+  } else if (isMaryBlock(block)) {
+    b = block.mary as Schema.BlockMary;
+  }
+  return b;
+};
+
+export const mapMetadata = (serverAssetMetadata: AssetMetadata): Asset.TokenMetadata => ({
+  decimals: serverAssetMetadata.decimals.value,
+  desc: serverAssetMetadata.description?.value,
+  icon: serverAssetMetadata.logo?.value,
+  name: serverAssetMetadata.name?.value,
+  ticker: serverAssetMetadata.ticker?.value,
+  url: serverAssetMetadata.url?.value
+  // Missing fields: sizedIcons
+});
+
+const mapAssets = (assets: { [k: string]: Schema.AssetQuantity } | undefined) =>
+  // Missing: transactionOutput, asset
```

Comment:
You're probably looking at the wrong type:
- `Token` represents value on utxo
- `Asset` is what's minted - it's what we want here

Reply:
Am I ? I intended to map the [Token](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/cardano-graphql-services/src/Schema/types/Transaction/Token.ts) type. I will rename this fc to `mapToken` to be more clear.
I think the best is to  just leave `tx.mint` and `output.value.assets`  empty for now with a comment that `assetId` is missing (to reference the asset edge).

Reply:
Ok, so the idea is in the first instance to insert the Asset object into this BlockHandler and then at AssetHandler update each of these asset objects with its metadata? In this case be must be sure that the block handler runs first.
So, the mint property would be necessary there, I will add this.
Not sure how the block handler interface will help

Reply:
> Ok, so the idea is in the first instance to insert the Asset object into this BlockHandler and then at AssetHandler update each of these asset objects with its metadata?

> So, the mint property would be necessary there, I will add this.

Something like that, but no, it doesn't sound right. The purpose of BlockHandlers is to decompose BlockBlockHandler and make it possible to create a database that has only a subset of data by using a subset of BlockHandlers. So I think the BlockBlockHandler is not supposed to write Assets at all.

> In this case be must be sure that the block handler runs first.

Original idea was to use [variables](https://github.com/input-output-hk/cardano-js-sdk/pull/197/files#diff-e9fbda3cd7bac9880ca294f5c1de0b2e102ac4a15a111f8799da73a1e2d66182R36-R37) to build a dependency tree: it should be possible to compute the correct order of handlers. Let's keep it out of scope of this PR and just put the handlers in order that we need.

> So, the mint property would be necessary there, I will add this.

I think BlockBlockHandler shouldn't insert assets. It's what AssetBlockHandler.

Reply:
Also I'm thinking that since we need to query the tip in order to start ChainFollower, maybe it would be best to write partial blocks as "core functionality" (like a hidden BlockHandler that always runs first). Thoughts @rhyslbw ?

Reply:
> Also I'm thinking that since we need to query the tip in order to start ChainFollower, maybe it would be best to write partial blocks as "core functionality" (like a hidden BlockHandler that always runs first).

Yes, this is the first handler to implement. The asset handler was being nominated first since it requires both the query and processing step, but it does need to come after a pure chain data handler to allow the sync state to be understood at startup.